File: code-examples/scala/authentication/x509.scala
import org.mongodb.scala.{MongoClient, MongoCommandException, Observable}
import scala.concurrent.Await
import scala.concurrent.duration.Duration

// Replace the following with values for your environment.
val clusterUrl = "<MongoDB cluster url>"
val clientPEMFile = "<path to the client pem certificate file>"
val authMechanism = "MONGODB-X509"

// Replace the following with your MongoDB deployment's connection string.
val uri = s"mongodb+srv://$clusterUrl/?authMechanism=$authMechanism&tls=true&tlsCertificateKeyFile=$clientPEMFile"

// Create a new MongoClient
val mongoClient: MongoClient = MongoClient(uri)

// Observable helper
def getResults[T](obs: Observable[T]): Seq[T] = {
  Await.result(obs.toFuture(), Duration(10, "second"))
}

// Function to connect to the server
def run(): Unit = {
  try {
    // Establish and verify connection
    val pingCommand = getResults(mongoClient.getDatabase("admin").runCommand(Document("ping" -> 1)))
    println("Connected successfully to server")
  } catch {
    case e: MongoCommandException => println("Failed to connect to server: " + e)
  } finally {
    // Ensures that the client will close when you finish/error
    mongoClient.close()
  }
}

run()

File: code-examples/php/authentication/x509.php
<?php
require_once __DIR__ . "/vendor/autoload.php";

// Replace the following with values for your environment.
$clusterUrl = "<MongoDB cluster url>";
$clientPEMFile = "<path to the client pem certificate file>";

$authMechanism = "MONGODB-X509";

// Replace the following with your MongoDB deployment's connection string.
$uri =
 "mongodb+srv://$clusterUrl/?authMechanism=$authMechanism&ssl=true&sslPEMKeyFile=$clientPEMFile";

// Create a new MongoDB\Client
$client = new MongoDB\Client($uri);

// Function to connect to the server
function run($client) {
  try {
    // Establish and verify connection
    $client->admin->command(['ping' => 1]);
    echo "Connected successfully to server";
  } catch (Exception $e) {
    echo $e;
  } finally {
    // Ensures that the client will close when you finish/error
    $client = null;
  }
}
run($client);
?>
